File: test_frontend.js
const { chromium } = require('playwright');

(async () => {
    const browser = await chromium.launch({ headless: true });
    const page = await browser.newPage();

    // We will do a simple manual trigger via playwright evaluate to test our new logic
    await page.goto('http://localhost:8080/index.html');

    // Wait for the app to init
    await page.waitForFunction(() => typeof window.addToMultivision === 'function');

    // Add some mocked multiview streams
    await page.evaluate(() => {
        // Force the app to add streams properly using its own function
        window.addToMultivision('a.com', 'Match A', '1');
        window.addToMultivision('b.com', 'Match B', '2');
        window.addToMultivision('c.com', 'Match C', '3');
    });

    await page.waitForTimeout(1000);

    // Trigger the events programmatically
    const result = await page.evaluate(() => {
        const cells = document.querySelectorAll('.mv-cell');
        if (cells.length < 3) return "Not enough cells: " + cells.length;

        // Drag cell 0 to cell 1
        window.draggedMvIdx = 0;

        // dispatch dragenter on cell 1
        const dragEnterEvent = new Event('dragenter');
        cells[1].dispatchEvent(dragEnterEvent);

        return window.mvFlux.map(s => s.name);
    });

    console.log("After drag enter (0 -> 1), expected [Match B, Match A, Match C]:", result);

    // Test if dragend cleans up
    const finalResult = await page.evaluate(() => {
        const cells = document.querySelectorAll('.mv-cell');

        const dragEndEvent = new Event('dragend');
        cells[0].dispatchEvent(dragEndEvent);

        return {
           flux: window.mvFlux.map(s => s.name),
           draggedMvIdx: window.draggedMvIdx
        };
    });
    console.log("After drag end, cleanup state:", finalResult);

    await browser.close();
})();

File: test_frontend.py
from playwright.sync_api import sync_playwright
import time

def run_cuj(page):
    page.on("console", lambda msg: print(f"Browser console: {msg.text}"))
    page.on("pageerror", lambda err: print(f"Browser error: {err}"))

    page.goto("http://localhost:3000/index.html")

    # Wait for the main elements to load
    page.wait_for_timeout(2000)

    page.evaluate("""
        window.hasLoadedOnce = true;
        if (document.getElementById('ov')) document.getElementById('ov').style.display = 'none';

        const mockMatches = [{
            "id": "mock_1",
            "homeTeam": "Mock Team A",
            "awayTeam": "Mock Team B",
            "league": "Mock League",
            "status": "live",
            "minute": "45'",
            "startTime": "20:00",
            "matchDate": "2026-05-26",
            "score": ["1", "0"],
            "streamLinks": [{"url": "http://example.com"}]
        }];

        if (window.S) {
            window.S.matches = mockMatches;
            if (typeof window.buildEPG === 'function') {
                window.buildEPG(mockMatches);
            } else if (typeof window.updateMatchUiAfterScrape === 'function') {
                window.updateMatchUiAfterScrape(mockMatches, true);
            }
        }
    """)
    page.wait_for_timeout(2000)
    page.screenshot(path="verification_screenshot2.png")

with sync_playwright() as p:
    browser = p.chromium.launch(headless=True)
    page = browser.new_page()
    run_cuj(page)
    browser.close()
